charts/linkerd-control-plane/values.yaml

```diff
@@ -286,10 +289,16 @@ proxy:
     collector:
       # -- The collector endpoint to send traces to.
       endpoint: ""
-      # -- The identity of the collector in the linkerd mesh. If the collector
-      # is unmeshed, this should remain unset.
+      # -- The identity of the collector in the linkerd mesh.
       meshIdentity:
+        # -- Service account name for the trace collector. If there's no explicitly set service account for the trace
+        # collector, this should be set to the name of the deployment/statefulset/daemonset of the trace collector
+        # instead.
         serviceAccountName: ""
+        # -- Service account namespace for the trace collector. If there's no explicitly set service account for the
+        # trace collector, this should be set to the namespace of the deployment/statefulset/daemonset of the trace
+        # collector instead.
+        serviceAccountNamespace: ""
```

Comment:
Prior to this change, did the namespace need to be specified as part of the `serviceAccountName`?  Is this breaking change necessary?

Furthermore, we're introducing a new, required value here so this breaking change will affect everyone who has `proxy.tracing` enabled.

Reply:
In that case, can we leave serviceAccountNamespace in the values.yaml but document it as deprecated and that the name/namespace fields are preferred but that it still exists for backwards compatibility.  Let's leave a paper trail of why there are two different configuration methods here.